packages/twenty-server/src/modules/calendar/services/google-calendar-sync/google-calendar-sync.service.ts

```diff
@@ -320,14 +321,30 @@ export class GoogleCalendarSyncService {
     let hasMoreEvents = true;
 
     while (hasMoreEvents) {
-      const googleCalendarEvents = await googleCalendarClient.events.list({
-        calendarId: 'primary',
-        maxResults: 500,
-        syncToken: emailOrDomainToReimport ? undefined : syncToken,
-        pageToken: nextPageToken,
-        q: emailOrDomainToReimport,
-        showDeleted: true,
-      });
+      const googleCalendarEvents = await googleCalendarClient.events
+        .list({
+          calendarId: 'primary',
+          maxResults: 500,
+          syncToken: emailOrDomainToReimport ? undefined : syncToken,
+          pageToken: nextPageToken,
+          q: emailOrDomainToReimport,
+          showDeleted: true,
+        })
+        .catch(async (error: GaxiosError) => {
+          if (error.response?.status !== 410) {
+            throw error;
+          }
+
+          await this.calendarChannelRepository.updateSyncCursor(
+            null,
+            connectedAccountId,
+            workspaceId,
+          );
+
+          throw new Error(
+            `Sync token no longer valid for connected account ${connectedAccountId} in workspace ${workspaceId}, resetting sync cursor.`,
```

Comment:
should we throw here or change the status to FAILED?

Reply:
We should just reset the sync token, without a sync token, the next sync will be a full sync

Reply:
so we don't throw at all in this case?